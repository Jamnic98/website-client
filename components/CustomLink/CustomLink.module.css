.customLink {
  height: auto;
  width: auto;
}