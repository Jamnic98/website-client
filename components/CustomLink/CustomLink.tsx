import Link from 'next/link';
import styles from './CustomLink.module.css';

interface Props {
  url: string;
  children?: any;
}

function CustomLink({ url, children }: Props) {
  return (
    <Link href={url}>
      <a className={styles.customLink}>{children}</a>
    </Link>
  );
}

export default CustomLink;
