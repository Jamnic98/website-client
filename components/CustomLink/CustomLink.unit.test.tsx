import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import CustomLink from './CustomLink';

describe('test CustomLink', () => {
  const testLink = (
    <CustomLink href='/'>
      <div data-testid='childElement'>home</div>
    </CustomLink>
  );
  it('displays correctly', () => {
    render(testLink);
    const customLink = screen.getByText('home');
    expect(customLink).toBeInTheDocument();
    const childElement = screen.getByTestId('childElement');
    expect(customLink).toContainElement(childElement);
  });

  it('renders CustomLink unchanged', () => {
    const testLink = (
      <CustomLink href='/'>
        <div data-testid='childElement'></div>
      </CustomLink>
    );
    const tree = renderer.create(testLink).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
