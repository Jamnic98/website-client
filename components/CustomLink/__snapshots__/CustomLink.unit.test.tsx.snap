// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CustomLink renders CustomLink unchanged 1`] = `
<a
  className="customLink"
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <div
    data-testid="childElement"
  />
</a>
`;
